src/lib/message/L2ToL1Message.ts

```diff
@@ -352,75 +358,85 @@ export class L2ToL1MessageReader extends L2ToL1Message {
   public async getFirstExecutableBlock(
     l2Provider: Provider
   ): Promise<BigNumber> {
-    throw new ArbTsError("getFirstExecutableBlock not implemented")
-    /*
-    // expected number of L1 blocks that it takes for an L2 tx to be included in a L1 assertion
-    const ASSERTION_CREATED_PADDING = 50
-    // expected number of L1 blocks that it takes for a validator to confirm an L1 block after the node deadline is passed
-    const ASSERTION_CONFIRMED_PADDING = 20
-
     // TODO: create version that queries multiple L2 to L1 txs, so a single multicall can make all requests
     // we assume the L2 to L1 tx is valid, but we could check that on the constructor that the L2 to L1 msg is valid
-    const network = await getL2Network(l2Provider)
+    const l2Network = await getL2Network(l2Provider)
 
-    // TODO: use IRollupUser interface instead
-    const rollup = RollupUserFacet__factory.connect(
-      network.ethBridge.rollup,
+    const rollup = new Contract(
+      l2Network.ethBridge.rollup,
+      [
+        'function getNode(uint64 nodeNum) public view override returns (tuple(bytes32 stateHash, bytes32 challengeHash, bytes32 confirmData, uint64 prevNum, uint64 deadlineBlock, uint64 noChildConfirmedBeforeBlock, uint64 stakerCount, uint64 childStakerCount, uint64 firstChildBlock, uint64 latestChildNumber, uint64 createdAtBlock, bytes32 nodeHash) node)',
+        'event NodeConfirmed(uint64 indexed nodeNum, bytes32 blockHash, bytes32 sendRoot)',
+      ],
       this.l1Provider
     )
 
-    const proof = await this.tryGetProof(l2Provider)
+    const proof = await this.getOutboxProof(l2Provider)
```

Comment:
I think this will throw if the node is not confirmed

Reply:
removed since we don't actually need that